server/rooms.ts

```diff
@@ -824,6 +829,26 @@ export abstract class BasicRoom {
 			}
 		}
 	}
+	sanitizeSection(section: string) {
+		const target = toID(section);
+		if (!RoomSections.sections.includes(target as any)) {
+			throw new Chat.ErrorMessage(`"${target}" is not a valid room section. Valid categories include: ${RoomSections.sections.join(', ')}`);
+		}
+		return target as RoomSection;
+	}
+	adjustSection(newSection: string) {
+		if (!this.persist) {
+			throw new Chat.ErrorMessage(`You cannot change the section of temporary rooms.`);
+		}
+		const section = this.sanitizeSection(newSection);
+		const oldSection = this.settings.section;
+		if (oldSection === section) {
+			throw new Chat.ErrorMessage(`${this.title}'s room section is already set to "${RoomSections.sectionNames[oldSection]}".`);
+		}
+		this.settings.section = this.section = section;
```

Comment:
Why does the section have to be included both in the settings and the `Room` object? One place should suffice.